crates/re_log_types/src/component_types/tensor.rs

```diff
@@ -399,50 +399,111 @@ impl Tensor {
         self.shape.as_slice()
     }
 
+    /// Returns the shape of the tensor with all trailing dimensions of size 1 ignored.
+    #[inline]
+    pub fn shape_short(&self) -> &[TensorDimension] {
+        self.shape
+            .iter()
+            .enumerate()
+            .rev()
+            .find(|(_, dim)| dim.size != 1)
+            .map_or(&self.shape[0..1], |(i, _)| &self.shape[..(i + 1)])
+    }
+
     #[inline]
     pub fn num_dim(&self) -> usize {
         self.shape.len()
     }
 
     /// If this tensor is shaped as an image, return the height, width, and channels/depth of it.
     pub fn image_height_width_channels(&self) -> Option<[u64; 3]> {
-        if self.shape.len() == 2 {
-            Some([self.shape[0].size, self.shape[1].size, 1])
-        } else if self.shape.len() == 3 {
-            let channels = self.shape[2].size;
-            // gray, rgb, rgba
-            if matches!(channels, 1 | 3 | 4) {
-                Some([self.shape[0].size, self.shape[1].size, channels])
-            } else {
-                None
+        let shape_short = self.shape_short();
+
+        match shape_short.len() {
+            1 => {
+                // Special case: Nx1 and Nx1x1 tensors are treated as Nx1 grey images.
```

Comment:
Why not `Nx1x1x1` tensors?

Reply:
I figured that goes to far.. but I suppose that would indeed be more consistent!